monitor/services/src/main/java/org/datacleaner/monitor/server/controllers/ResultsFolderController.java

```diff
@@ -49,24 +51,46 @@
     @RolesAllowed(SecurityRoles.VIEWER)
     @RequestMapping(method = RequestMethod.GET, produces = "application/json")
     @ResponseBody
-    public List<Map<String, String>> resultsFolderJson(@PathVariable("tenant") final String tenant) {
-        final TenantContext context = _tenantContextFactory.getContext(tenant);
+    public List<Map<String, String>> resultsFolderJson(@PathVariable("tenant") final String tenant,
+            @RequestParam(value = "not_before", required = false) final String timestamp) {
 
+        final TenantContext context = _tenantContextFactory.getContext(tenant);
         final RepositoryFolder resultsFolder = context.getResultFolder();
-
         final List<Map<String, String>> result = new ArrayList<>();
 
         {
             final List<RepositoryFile> files =
                     resultsFolder.getFiles(null, FileFilters.ANALYSIS_RESULT_SER.getExtension());
             for (final RepositoryFile file : files) {
                 final Map<String, String> map = new HashMap<>();
-                map.put("filename", file.getName());
+                final String name = file.getName();
+                if (!isEligibleJob(timestamp, name)) {
```

Comment:
Still not important, but this could easily be reverted to a normal flow now:
```java
if (isEligibleJob(timestamp, name)) {
    map.put("filename", name);
    map.put("repository_path", file.getQualifiedPath());
    result.add(map)
}
```